kpops/components/common/kubernetes_model.py

```diff
@@ -42,6 +42,78 @@ class ImagePullPolicy(str, enum.Enum):
     NEVER = "Never"
 
 
+Weight = Annotated[int, pydantic.Field(ge=1, le=100)]
+
+
+class Operator(str, enum.Enum):
+    IN = "In"
+    NOT_IN = "NotIn"
+    EXISTS = "Exists"
+    DOES_NOT_EXIST = "DoesNotExist"
+
+
+class SelectorRequirement(DescConfigModel, CamelCaseConfigModel):
+    key: str
+    operator: Operator
+    values: list[str] | None = pydantic.Field(
+        default=None,
+        description="An array of string values. If the operator is In or NotIn, the values array must be non-empty. If the operator is Exists or DoesNotExist, the values array must be empty. If the operator is Gt or Lt, the values array must have a single element, which will be interpreted as an integer. This array is replaced during a strategic merge patch.",
+    )
+
+
+class NodeSelectorTerm(DescConfigModel, CamelCaseConfigModel):
+    match_expressions: list[SelectorRequirement] | None = None
+    match_fields: list[SelectorRequirement] | None = None
+
+
+class NodeSelector(DescConfigModel, CamelCaseConfigModel):
+    node_selector_terms: list[NodeSelectorTerm]
+
+
+class PreferredSchedulingTerm(DescConfigModel, CamelCaseConfigModel):
+    preference: NodeSelectorTerm
+    weight: Weight
+
+
+class NodeAffinity(DescConfigModel, CamelCaseConfigModel):
+    required_during_scheduling_ignored_during_execution: NodeSelector | None = None
+    preferred_during_scheduling_ignored_during_execution: (
+        list[PreferredSchedulingTerm] | None
+    ) = None
+
+
+class LabelSelector(DescConfigModel, CamelCaseConfigModel):
+    match_labels: dict[str, str] | None = None
+    match_expressions: list[SelectorRequirement] | None = None
+
+
+class PodAffinityTerm(DescConfigModel, CamelCaseConfigModel):
+    label_selector: LabelSelector | None = None
+    topology_key: str
+    namespaces: list[str] | None = None
+    weight: int
```

Comment:
Is there any reason the `weight` here is `int` and not of type `Weight`?

Reply:
well-spotted 😉